js-project-guidelines.md

```diff
@@ -1,113 +1,147 @@
+- **Browser compatibility**, with the possible exceptions being:
+  - Access to the file system.
+  - Native bindings.
+  - Network transports (uTP, udt, curveCP, etc) that are not available in the browser.
+- **Don't break CommonJS's** `require`. This means that if someone requires a JavaScript module from the IPFS ecosystem, they should be able to require it and use browserify, webpack or other bundlers without having to worry about adding special shims for module internals.
+- **Encourage Contribution**.
+- **Great UX** for everyone involved.
 
-- project stucture.
-- code style.
-- continuous integration.
-- tests.
-- tasks (asset pipeline, transpiling, releasing, etc).
-- dependency management.
+## The Guidelines
 
-Our tool picks for each of this dimensions are not set in stone, nor do we plan in any way of halting our search to improve them.
-
-#### a) Linting & Code Style
+#### Linting & Code Style
 
 IPFS JavaScript projects default to [standard](https://github.com/feross/standard) code style. It is a great and clean codestyle and its adoption is increasing significantly, making the code that we write familiar to the majority of the developers.
 
-However we've added an extra linting rule hto enforce the use of [strict mode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode) to avoid issues we had when using ES2015 features outside of strict mode. We enforce this rule by using [eslint](http://eslint.org/) and extending [standard module](https://github.com/feross/standard) with the [eslint-config-standard](https://github.com/feross/eslint-config-standard).
+However, we've added an extra linting rule: Enforce the use of [strict mode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode). This avoids issues we had when using ES2015 features outside of strict mode. We enforce this rule by using [eslint](http://eslint.org/) and extending [standard module](https://github.com/feross/standard) with the [eslint-config-standard](https://github.com/feross/eslint-config-standard).
 
-#### b) Test
+#### Test
 
-Since `js-ipfs` is meant to be both a Node.js and Browser app, we strongly recommend to have tests that run in both platforms, always. For most cases, we use [mocha](http://mochajs.org) to run write the tests and [karma](http://karma-runner.github.io) to automate the test execution in the browser. This solution has been extremely convinient.
+Since `js-ipfs` is meant to be both a Node.js and Browser app, we strongly recommend having tests that run in both platforms, always. For most cases, we use [mocha](http://mochajs.org) to run write the tests and [karma](http://karma-runner.github.io) to automate the test execution in the browser. This solution has been extremely convenient.
 
-#### c) Build
+#### Build
```

Comment:
Grammar nit: if we're using "Testing" in the heading above, all of these should be in the progressive tense too, right?